modules/frontend/search_sharder.go

```diff
@@ -201,18 +202,17 @@ func (s *asyncSearchSharder) backendRequests(ctx context.Context, tenantID strin
 // that covers the ingesters. If nil is returned for the http.Request then there is no ingesters query.
 // since this function modifies searchReq.Start and End we are taking a value instead of a pointer to prevent it from
 // unexpectedly changing the passed searchReq.
-func (s *asyncSearchSharder) ingesterRequest(ctx context.Context, tenantID string, parent *http.Request, searchReq tempopb.SearchRequest) (*http.Request, error) {
+func (s *asyncSearchSharder) ingesterRequests(ctx context.Context, tenantID string, parent *http.Request, searchReq tempopb.SearchRequest, reqCh chan *http.Request) error {
 	// request without start or end, search only in ingester
 	if searchReq.Start == 0 || searchReq.End == 0 {
-		return buildIngesterRequest(ctx, tenantID, parent, &searchReq)
+		return buildIngesterRequest(ctx, tenantID, parent, &searchReq, reqCh)
```

Comment:
this will still build one request, should be okay imo but wanted to point it out.

Reply:
this is from past, where we allowed requests without start and end.

Reply:
Good call out.  I think let's follow up with a breaking change to validate that we pass a start/end and remove this chunk.